### PR TITLE
batch delete, auth middleware, requests & some constants 

### PR DESCRIPTION
**Pull request by yash-dxt**

### CODE REVIEW

File: lambda/invoke.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+	if err != nil {
+		return res, err
+	}
+
+	return convertServiceResponse[I](m["body"])
+}
+
+func convertServiceResponse[I any](serviceRes interface{}) (I, error) {
+	s := fmt.Sprint(serviceRes)
```
Comment (awenborar) [inline lambda/invoke.go:70]: why do we need this conversion here?

File: utils/get_user_id_from_auth_service.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/metaphi-org/go-infra-sdk/lambda"
+)
+
+const auth_service_userid_endpoint = "/auth/userId"
+
+type CheckSessionAuthRequest struct {
+	UserId string `json:"userId"`
+}
+
+func GetUserIdFromAuthService(ctx context.Context, config aws.Config, authServiceLambdaName string, sessionId string) (string, error) {
```
Comment (awenborar) [inline utils/get_user_id_from_auth_service.go:18]: if the function is specifically to talk to auth service, why does it need to take the lambda name as input?

Comment (yash-dxt) [reply utils/get_user_id_from_auth_service.go:18]: auth service/auth lambda can have different names for different environments?

**Comment by awenborar** [comment]
Sdk can take care of different environments?

File: utils/get_user_id_from_auth_service.go (hunk 2 above)
Comment (awenborar) [reply utils/get_user_id_from_auth_service.go:18]: Sdk can take care of different environments?

File: lambda/invoke.go (hunk 1 above)
Comment (yash-dxt) [reply lambda/invoke.go:70]: should be done here right? if we do it elsewhere we'll be repeating code

**Review by awenborar — APPROVED** [review]

**Pull request merged**